eras/shelley/impl/src/Cardano/Ledger/Shelley/Rules/Ledger.hs

```diff
@@ -98,6 +100,16 @@ deriving instance Eq (PParams era) => Eq (LedgerEnv era)
 instance NFData (PParams era) => NFData (LedgerEnv era) where
   rnf (LedgerEnv _slotNo _ix pp _account _mempool) = rnf pp
 
+instance EraPParams era => EncCBOR (LedgerEnv era) where
+  encCBOR LedgerEnv {..} =
```

Comment:
I suggest adding pattern match here, so we don't forget to change this instance when or if `LedgerEnv` changes:
```suggestion
  encCBOR env@LedgerEnv {..} =
    let LedgerEnv _ _ _ _ _ = env in
```

Reply:
This triggers the `unused-pattern-binds` warning unfortunately.

Reply:
You could do this instead, without warnings:
```suggestion
  encCBOR env@(LedgerEnv _ _ _ _ _) =
    let LedgerEnv {..} = env in
```

Reply:
Thank you @teodanciu I forgot the order in which we usually do this :smile: